server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -305,10 +309,87 @@ public void onFailure(Exception e) {
         }
     }
 
-    private void failDecommissionedNodes(ClusterState state) {
+    private void drainNodesWithDecommissionedAttribute(ClusterState state, DecommissionRequest decommissionRequest) {
+        decommissionController.updateMetadataWithDecommissionStatus(DecommissionStatus.DRAINING, new ActionListener<>() {
+            @Override
+            public void onResponse(DecommissionStatus status) {
+                logger.info("updated the decommission status to [{}]", status);
+                // set the weights
+                setRoutingWeightsToAwarenessAttribute(decommissionRequest);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error(
+                    () -> new ParameterizedMessage(
+                        "failed to update decommission status for attribute [{}] to [{}]",
+                        decommissionRequest.getDecommissionAttribute().toString(),
+                        DecommissionStatus.DRAINING
+                    ),
+                    e
+                );
+                // since we are not able to update the status, we will clear the voting config exclusion we have set earlier
+                clearVotingConfigExclusionAndUpdateStatus(false, false);
+            }
+        });
+    }
+
+    void setRoutingWeightsToAwarenessAttribute(DecommissionRequest decommissionRequest) {
+        ClusterState state = clusterService.getClusterApplierService().state();
+
+        DecommissionAttributeMetadata decommissionAttributeMetadata = state.metadata().decommissionAttributeMetadata();
+        if (decommissionAttributeMetadata == null) {
+            clearVotingConfigExclusionAndUpdateStatus(false, false);
+            return;
+        }
+        assert decommissionAttributeMetadata.status().equals(DecommissionStatus.DRAINING)
+            : "Unexpected status encountered while decommissioning nodes.";
+        DecommissionAttribute decommissionAttribute = decommissionAttributeMetadata.decommissionAttribute();
+        List<String> awarenessAttributeValues = forcedAwarenessAttributes.get(decommissionAttribute.attributeName());
+
+        Map<String, Double> weights = new HashMap<>();
+        awarenessAttributeValues.forEach(awarenessValue -> {
+            if (awarenessValue.equalsIgnoreCase(decommissionAttribute.attributeValue())) {
+                weights.put(awarenessValue, Double.valueOf(0.0));
+            } else {
+                weights.put(awarenessValue, Double.valueOf(1.0));
+            }
+        });
```

Comment:
what if the weights were 3:2:1 to start with would that be reset back?

Reply:
@Bukhtawar .. Currently we are not setting the weights back to original value. We are leaving currently it to same value and rely the client to set the appropriate weights back. 
I am opening a task on discussing the same. 

https://github.com/opensearch-project/OpenSearch/issues/4803

Reply:
why can't we do 3:2:0 in the above example

Reply:
Can we do 3:2:0 instead of 1:1:0?

Reply:
@Bukhtawar .. Not sure if i have understood this.. Are you saying to set different weights for the attributes? And how to decide which attribute should have 3 or 2. This can lead to unbalanced traffic movement.

Reply:
@pranikum Consider this example -
Let's say before decommission the weights were z1 -> 3, z2 -> 2, z3 -> 1

Now as per the current implementation, when we decommission z3, we would set the weights to z1 -> 1, z2 -> 1, z3 -> 0. @Bukhtawar's point is rather than doing this, only set the decommissioned zone weight to 0 and leave other zone's weight untouched. Something like, z1-> 3, z2 -> 2, z1 -> 0

But this again has a caveat to it when a different non decommissioned zone already had weight 0 (z1 -> 0, z2 -> 2, z3 -> 1) and if we decommission z3 and set weight to (z1 -> 0, z2 -> 2, z3 -> 0) this again can be problem.

Now, we don't know why the user has set weight for z1 as 0 but again we cannot let weights of 2 zones to be 0.

Reply:
Ok.. I understood the situation now. But this can lead to 2 zone with weights 0. If we go by this flow as well.
Initial weights : z1 -> 3 z2 -> 2 z3 -> 1
Now lets say we have these actions 
1. Decommission z3 - (Weights after this step z1 : 3 z2: 2 z3: 0) 
2. Recommission z3 - (Weights after this step z1 : 3 z2: 2 z3: 0) (Since currently we are not setting weights back) 
3. Decommission z1 - (Weights after this step z1 : 0 z2: 2 z3: 0)  Here we have 2 decommissioned zone.
4. Recommission z1 - (Weights after this step z1 : 0 z2: 2 z3: 0) 

I feel we should do this after we have done https://github.com/opensearch-project/OpenSearch/issues/4803
In this case weights will be set back during recommission and will not lead to multiple weights being 0.

Reply:
For this case, I think we should let the weights of not decommissioned zone as is. If a non decommissioned zone had weight zero then we can assign the weight of decommissioned zone to it, and put weight of decommissioned zone to zero
@Bukhtawar thoughts on this?

Reply:
Should we have a mechanism to set weights as -1 for decommission to distinguish from an explicit weight as 0. That way we reject the second decommission call if weights are unspecified(-1)

Reply:
As disscussed. We will not be setting weights for now. We will let the user to set the weights.